data/group/src/main/java/com/kappzzang/jeongsan/api/GroupRetrofitService.kt

```diff
@@ -0,0 +1,59 @@
+package com.kappzzang.jeongsan.api
+
+import com.kappzzang.jeongsan.entity.CompleteGroupResponse
+import com.kappzzang.jeongsan.entity.CreateGroupResponse
+import com.kappzzang.jeongsan.entity.GetGroupResponse
+import com.kappzzang.jeongsan.entity.GetLinkResponse
+import com.kappzzang.jeongsan.entity.GetMemberInfoResponse
+import com.kappzzang.jeongsan.entity.GetMyExpenseResponse
+import com.kappzzang.jeongsan.entity.JoinGroupResponse
+import retrofit2.Response
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.PATCH
+import retrofit2.http.POST
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+interface GroupRetrofitService {
+    @GET("/api/teams")
+    suspend fun getGroupInfo(
+        @Header("Authorization") token: String,
+        @Query("isClosed") isCompleted: Boolean
+    ): Response<GetGroupResponse>
+
+    @POST("/api/teams")
+    suspend fun createGroup(
+        @Header("Authorization") token: String,
+        @Query("name") name: String,
+        @Query("subject") subject: String,
+        @Query("members") memberIdList: List<Long>
+    ): Response<CreateGroupResponse>
+
+    @PATCH("/api/teams/{teamId}")
+    suspend fun completeGroup(
+        @Header("Authorization") token: String,
+        @Path("teamId") groupId: Long
+    ): Response<CompleteGroupResponse>
+
+    @GET("/api/teams/{teamId}/members")
+    suspend fun getMemberInfo(
+        @Header("Authorization") token: String,
+        @Path("teamId") groupId: Long
+    ): Response<GetMemberInfoResponse>
+
+    @POST("/api/members/join/{teamId}")
+    suspend fun joinGroup(
+        @Header("Authorization") token: String,
+        @Path("teamId") groupId: Long,
+        @Body memberId: Long
+    ): Response<JoinGroupResponse>
```

Comment:
P3: Body가 하나뿐이라서 이렇게 진행해도 무방할 것 같지만, 전체적인 일관성을 고려해 Data Class를 만들어서 구현하는것도 괜찮을 것 같아요!

```suggestion
    @POST("/api/members/join/{teamId}")
    suspend fun joinGroup(
        @Header("Authorization") token: String,
        @Path("teamId") groupId: Long,
        @Body request: JoinGroupRequest
    ): Response<JoinGroupResponse>

data class JoinGroupRequest(
    @SerializedName("memberId")
    val memberId: Long
)
```